include/extractor/guidance/classification_data.hpp

```diff
@@ -18,63 +15,129 @@ namespace extractor
 namespace guidance
 {
 
-enum class FunctionalRoadClass : std::uint8_t
+struct RoadClassification
 {
-    UNKNOWN = 0,
-    MOTORWAY,
-    MOTORWAY_LINK,
-    TRUNK,
-    TRUNK_LINK,
-    PRIMARY,
-    PRIMARY_LINK,
-    SECONDARY,
-    SECONDARY_LINK,
-    TERTIARY,
-    TERTIARY_LINK,
-    UNCLASSIFIED,
-    RESIDENTIAL,
-    SERVICE,
-    LIVING_STREET,
-    LOW_PRIORITY_ROAD // a road simply included for connectivity. Should be avoided at all cost
-};
+    // a class that behaves like a motorway (separated directions)
+    bool motorway_class : 1;
```

Comment:
Bitfields and bool's are apparently not cross-platform compatible (read: non-gcc this will break). The easiest solution that I have seen is to use `std::uint32_t` for all members that get packed into a 32bit block.